### PR TITLE
Toggle between windowed/fullscreen on desktop

### PR DESCRIPTION
**Pull request by pyrosphere**
On desktop it's currently possible to create a window either fullscreen or windowed.

This PR adds the methods to GLViewImpl on desktop to toggle between fullscreen and windowed. It includes a test for these methods.

### CODE REVIEW

File: cocos/platform/desktop/CCGLViewImpl-desktop.h [hunk 1]
```diff
@@ -79,6 +79,14 @@ class CC_DLL GLViewImpl : public GLView
     void pollEvents() override;
     GLFWwindow* getWindow() const { return _mainWindow; }
 
+    bool isFullscreen();
```
Comment (minggo) [inline cocos/platform/desktop/CCGLViewImpl-desktop.h]: bool isFullscreen() -> bool isFullscreen() const

File: cocos/platform/desktop/CCGLViewImpl-desktop.h [hunk 2]
```diff
@@ -79,6 +79,14 @@ class CC_DLL GLViewImpl : public GLView
...
+    void setFullscreen();
+    void setFullscreen(int monitorIndex);
+    void setFullscreen(const GLFWvidmode &videoMode, GLFWmonitor *monitor);
+    void setWindowed(int width, int height);
+    int getMonitorCount();
+    Size getMonitorSize();
```
Comment (minggo) [inline cocos/platform/desktop/CCGLViewImpl-desktop.h]: add const for `getMonitorCount` and `getMonitorSize` is better

File: tests/cpp-tests/Classes/WindowTest/WindowTest.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    WindowTest::onEnter();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
+    GLViewImpl* view = (GLViewImpl*)Director::getInstance()->getOpenGLView();
+    view->setFullscreen(1);
+#endif
+}
+
+std::string WindowTestFullscreen2::subtitle() const {
+    return "Fullscreen Monitor 2 (if present)";
+}
```
Comment (minggo) [inline tests/cpp-tests/Classes/WindowTest/WindowTest.cpp:54]: how about add the macro

```
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
```

like [this](https://github.com/cocos2d/cocos2d-x/blob/v3/tests/cpp-tests/Classes/controller.cpp#L39). Then the test case only appeared on these platforms, and don't have to add macro in this file.

File: cocos/platform/desktop/CCGLViewImpl-desktop.h [hunk 4]
```diff
@@ -79,6 +79,14 @@ class CC_DLL GLViewImpl : public GLView
     void pollEvents() override;
     GLFWwindow* getWindow() const { return _mainWindow; }
 
+    bool isFullscreen();
+    void setFullscreen();
+    void setFullscreen(int monitorIndex);
+    void setFullscreen(const GLFWvidmode &videoMode, GLFWmonitor *monitor);
```
Comment (minggo) [inline cocos/platform/desktop/CCGLViewImpl-desktop.h:85]: why need these two functions:
- void setFullscreen(int monitorIndex);
- void setFullscreen(const GLFWvidmode &videoMode, GLFWmonitor *monitor);

when will use these two functions?

Comment (pyrosphere) [reply cocos/platform/desktop/CCGLViewImpl-desktop.h:85]: The method `setFullscreen(int monitorIndex)` lets the game dev add options to switch the game to another monitor, if more than one is present. I personally use it along with `getMonitorCount()` for this.

The other method `setFullscreen(const GLFWvidmode &videoMode, GLFWmonitor *monitor);` is present to give game devs the parallel of `createWithFullScreen(const std::string& viewName, const GLFWvidmode &videoMode, GLFWmonitor *monitor)` which some might be using when creating the window.

**Comment by minggo** [comment]
the demo looks good, thanks

**Pull request merged**